src/Dependencies.jl

```diff
@@ -91,14 +91,25 @@ end
 
 # Add JSON serialization of dependencies
 string_or_nothing(x) = isnothing(x) ? x : string(x)
+
+major(v::VersionNumber) = v.major
+minor(v::VersionNumber) = v.minor
+patch(v::VersionNumber) = v.patch
+major(v::Pkg.Types.VersionBound) = v.t[1]
+minor(v::Pkg.Types.VersionBound) = v.t[2]
+patch(v::Pkg.Types.VersionBound) = v.t[3]
+__version(v::VersionNumber) = v
+__version(v::Pkg.Types.VersionSpec) = v.ranges[1].lower
+version(d::Union{Dependency, BuildDependency}) = __version(d.pkg.version)
```

Comment:
Minor nitpick:
```suggestion
version(d::AbstractDependency) = __version(getpkg(d).version)
```

So what happened is that in Julia v1.4 `d.pkg.version` is `VersionNumber` instead of `Pkg.Types.VersionBound` and this changed between v1.4-rc1 and v1.4?

Maybe add a test in [`test/declarative.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/master/test/declarative.jl)?